### PR TITLE
Allow users to configure `host`, `port` and `timeout` for build in TCP adapters

### PR DESCRIPTION
**Pull request by RemcoSmitsDev**
This PR enhances user configurability for built-in adapters utilizing TCP transport by allowing customization of `host`, `port`, and `timeout` settings. Additionally, we've implemented a feature where, if no **port** is specified, the system will automatically attempt to locate an **available port** on the provided **host**.

TODO:
- [x] Update or remove the cached binaries functionality, as the host/port information is no longer stored within the DebugAdapterBinary

### CODE REVIEW

File: crates/dap/src/transport.rs [hunk 1]
```diff
@@ -361,27 +361,38 @@ pub trait Transport: 'static + Send + Sync {
...
-    async fn get_open_port(host: Ipv4Addr) -> Option<u16> {
-        Some(
-            TcpListener::bind(SocketAddrV4::new(host, 0))
-                .await
-                .ok()?
-                .local_addr()
-                .ok()?
-                .port(),
-        )
+    pub async fn port(host: &TCPHost) -> Result<u16> {
```
Comment (Anthony-Eid) [inline crates/dap/src/transport.rs:386]: I might be wrong, but we're binding an open port given to us by the OS. Then we telling a debug adapter to connect to that port that's already being used, couldn't that cause an error in some cases?

File: crates/task/src/debug_format.rs [hunk 2]
```diff
@@ -16,13 +16,26 @@ impl Default for DebugConnectionType {
...
     pub port: Option<u16>,
     /// The host that the debug adapter is listening too
+    ///
+    /// Default: 127.0.0.1
     pub host: Option<Ipv4Addr>,
-    /// The max amount of time to connect to a tcp DAP before returning an error
+    /// The max amount of time in milliseconds to connect to a tcp DAP before returning an error
+    ///
+    /// Default: 2000ms
     pub timeout: Option<u64>,
```
Comment (Anthony-Eid) [inline crates/task/src/debug_format.rs:29]: We should probably make timeout a user setting too

File: crates/dap/src/transport.rs (hunk 1 above)
Comment (RemcoSmitsDev) [reply crates/dap/src/transport.rs:386]: No, we are binding here, but that does not mean we picked the spot for the debug adapter to starts it's tcp server for that port.

File: crates/task/src/debug_format.rs (hunk 2 above)
Comment (RemcoSmitsDev) [reply crates/task/src/debug_format.rs:29]: Should that override evrything or should that be the default/fallback value that is now 2000ms?

**Pull request merged**